applications/GeoMechanicsApplication/custom_processes/reset_displacement_process.cpp

```diff
@@ -0,0 +1,75 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+#include "reset_displacement_process.h"
+#include "includes/constitutive_law.h"
+#include "includes/initial_state.h"
+#include "includes/kratos_parameters.h"
+#include "includes/model_part.h"
+#include "includes/ublas_interface.h"
+#include "includes/variables.h"
+
+#include <vector>
+
+namespace Kratos
+{
+
+ResetDisplacementProcess::ResetDisplacementProcess(ModelPart& rModelPart, const Parameters&)
+    : mrModelPart(rModelPart)
+{
+}
+
+void ResetDisplacementProcess::ExecuteInitialize()
+{
+    block_for_each(mrModelPart.Elements(), [this](Element& rElement) {
+        std::vector<Vector> stresses_on_integration_points;
+        rElement.CalculateOnIntegrationPoints(PK2_STRESS_VECTOR, stresses_on_integration_points,
+                                              mrModelPart.GetProcessInfo());
+        std::vector<ConstitutiveLaw::Pointer> constitutive_laws;
+        rElement.CalculateOnIntegrationPoints(CONSTITUTIVE_LAW, constitutive_laws, mrModelPart.GetProcessInfo());
+
+        CheckRetrievedElementData(constitutive_laws, stresses_on_integration_points, rElement);
+
+        for (auto i = std::size_t{0}; i < constitutive_laws.size(); ++i) {
+            auto p_initial_state = make_intrusive<InitialState>();
+            p_initial_state->SetInitialStressVector(stresses_on_integration_points[i]);
+            p_initial_state->SetInitialStrainVector(Vector{ZeroVector{stresses_on_integration_points[i].size()}});
```

Comment:
This is the line that fixed the test, I'd like to discuss if this is correct and also whether we accept that if we don't add this strain, there is a hard crash (however, avoiding this probably means we need changes in structural, or even core.